### ISSUE TITLE
Strange behavior when quick replace is made

### ISSUE THREAD

**Issue by syru88** [arkivanov/Decompose#792]
[Source code](https://github.com/syru88/decompose-animation/tree/master)

I have following structure:
```
                     RootComponent
                      /          \
         MainComponent       WelcomeComponent
            /
InnerComponent
```

I use `replaceAll` to switch between `Main` and `Welcome`. `Main` has a child component `InnerComponent`. Initially `Main` is shown. When "Show Welcome screen" button is clicked, `Main` is replaced by `Welcome` and `InnerComponent` is destroyed. `Welcome` is shown just for a short period of time (less than animation time) and then is [closed](https://github.com/syru88/decompose-animation/blob/20d2d1bf62d3c28ae23f53a483aa1808b808f859/shared/src/commonMain/kotlin/com/example/myapplication/shared/welcome/WelcomeComponent.kt#L23-L23) (replaced by `Main`). `Main` and `Inner` are "freshly" created, but `InnerContent` isn't "refreshed" and keeps reference to old/destroyed `InnerComponent`.
[short_loading.webm](https://github.com/user-attachments/assets/a158516b-0cfe-4e6d-a232-b7ed79172f83)

It works correctly with longer delay.
[long_loading.webm](https://github.com/user-attachments/assets/9ba3572e-b1e8-471d-937b-81d698df3fbb)

**Comment by arkivanov** [comment]
This should work fine, as the navigation is performed synchronously. Unfortunately, the link you provided is broken, I could check that. It would be nice to have a reproducer code.

**Comment by syru88** [comment]
Agh, sorry, the repository was private 🤦  it's public now. I forgot to mention that I use `stackAnimation()` in `Children` function.

**Comment by arkivanov** [comment]
The animation shouldn't affect the navigation in any way. That's completely separate, just presentation.

**Comment by jaromirbarta** [comment]
In this use case, animation in **Children** view holds the child from the first instance of `InnerComponent` because the configuration (`data object InnerMain`) in a stack of the second instance of `InnerComponent` has an equal value.

**Comment by arkivanov** [comment]
Thanks! This turned out to be a bug, I will fix it.

**Comment by syru88** [comment]
Awesome, thank you very much! 👏

**Comment by arkivanov** [comment]
Should be fixed in version [3.2.0-beta03](https://github.com/arkivanov/Decompose/releases/tag/3.2.0-beta03).

**Comment by syru88** [comment]
Thank you, it works as expected 👍
